### PR TITLE
ShakeWake: Fixed instant wake after sleep issue in certain positions

### LINKED ISSUE #927: Hardware button to put watch to sleep causes instant wake under specific circumstances
### Verification

- [X] I searched for similar bug reports and found none was relevant.

### What happened?

Pressing the hardware button while holding the watch vertically and with shake to wake enabled with a low sensitivity (<50%) causes the watch to immediately wake up.

### What should happen instead?

_No response_

### Reproduction steps

- hold the watch vertically (with the hardware button facing up or down)
- press the hardware button on the watch face as if you were putting it to sleep

### More details?

- setting the sensitivity to over 50% prevents this issue
- lower sensitivities can cause the issue to occur at less extreme angles (closer to being held horizontally)

### Version

v1.8.0

### Companion app

Gadgetbridge 0.64.0

### PR DESCRIPTION
**Pull request by ic-27**
Added lastX var to track the previous x acceleration for correct calculation of speed. 
Closes #927.

### CODE REVIEW

**Comment by FintasticMan** [comment]
This looks good to me, but seeing as @geekbozu wrote the shake wake algorithm, I'd like to hear what they have to say.

File: src/components/motion/MotionController.cpp [hunk 1]
```diff
@@ -53,8 +54,7 @@ bool MotionController::ShouldRaiseWake(bool isSleeping) {
 
 bool MotionController::ShouldShakeWake(uint16_t thresh) {
   /* Currently Polling at 10hz, If this ever goes faster scalar and EMA might need adjusting */
-  int32_t speed = std::abs(z + (y / 2) + (x / 4) - lastY / 2 - lastZ) / (time - lastTime) * 100;
-  //(.2 * speed) + ((1 - .2) * accumulatedSpeed);
```
Comment (FintasticMan) [inline src/components/motion/MotionController.cpp:57]: And I don't think you intended to remove this comment.

Comment (isaacc27) [reply src/components/motion/MotionController.cpp:57]: Good catch, I must have accidentally deleted it when debugging.

**Review by FintasticMan — APPROVED** [review]
I think these changes are good. I would suggest using `shakewake` or `ShakeWake` as the module name in the commit, to show that it only affects shake to wake and not raise to wake.

**Review by JF002 — APPROVED** [review]

**Pull request merged**